pkg/transportrequest/gitutils_test.go

```diff
@@ -0,0 +1,175 @@
+	for {
+		c, err := iter.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return err
+		}
+
+		err = cb(c)
+		if err == storer.ErrStop {
+			break
+		}
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (iter *commitIteratorMock) Close() {
+
+}
+
+func TestRetrieveLabelStraightForward(t *testing.T) {
+
+	t.Run("single commit tests", func(t *testing.T) {
+
+		runTest := func(testConfig []string) {
+			t.Run(testConfig[0], func(t *testing.T) {
+				commitIter := &commitIteratorMock{
+					commits: []object.Commit{
+						object.Commit{
+							Hash:    plumbing.NewHash("3434343434343434343434343434343434343434"),
+							Message: testConfig[1],
+						},
+					},
+				}
+				labels, err := FindLabelsInCommits(commitIter, "TransportRequest")
+				if assert.NoError(t, err) {
+					expected := testConfig[2:]
+					if assert.Len(t, labels, len(expected)) {
+						assert.Subset(t, expected, labels)
+					}
+				}
+			})
+		}
+
+		tests := [][]string{
+			[]string{
+				"straight forward",
+				"this is a commit with TransportRequestId\n\nThis is the first line of the message body\nTransportRequest: 12345678",
+				"12345678",
+			},
+			[]string{
+				"trailing spaces after our value",
+				"this is a commit with TransportRequestId\n\nThis is the first line of the message body\nTransportRequest: 12345678  ",
+				"12345678",
+			},
+			[]string{
+				"trailing text after our value",
+				"this is a commit with TransportRequestId\n\nThis is the first line of the message body\nTransportRequest: 12345678 aaa",
+			},
+			[]string{
+				"leading whitespace before our label",
+				"this is a commit with TransportRequestId\n\nThis is the first line of the message body\n   TransportRequest: 12345678",
+				"12345678",
+			},
+			[]string{
+				"leading text before our label",
+				"this is a commit with TransportRequestId\n\nThis is the first line of the message body\naaa TransportRequest: 12345678",
+			},
+			[]string{
+				"whitespaces before column",
+				"this is a commit with TransportRequestId\n\nThis is the first line of the message body\nTransportRequest  : 12345678",
+				"12345678",
+			},
+			[]string{
+				"no whitespaces after column",
+				"this is a commit with TransportRequestId\n\nThis is the first line of the message body\nTransportRequest  :12345678",
+				"12345678",
+			},
+			[]string{
+				"two times the same id in the same commit",
+				"this is a commit with TransportRequestId\n\nThis is the first line of the message body\nTransportRequest : 12345678\nTransportRequest : 12345678",
+				"12345678",
+			},
+			[]string{
+				// we report the ids, this is basically an invalid state, but needs to be filtered out by the caller
+				"two different ids in the same commit",
+				"this is a commit with TransportRequestId\n\nThis is the first line of the message body\nTransportRequest : 12345678\nTransportRequest : 87654321",
+				"12345678", "87654321",
+			},
```

Comment:
well, I dont know the real commit message, so out of curiosity: true or false?
`"this is a commit with TransportRequestId\n\nThis is the first line of the message body\nTransportRequest: 12345678\nAnother line",
`

Reply:
... true, that would be a valid commit message containing a transport request id.